packages/eslint-plugin-baseui/src/no-deep-imports.js

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright (c) 2018-2020 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+module.exports = {
+  create(context) {
+    return {
+      ImportDeclaration(node) {
+        const match = node.source.value.match(/baseui\/(\w|-)*\//);
+        if (match) {
+          context.report({
+            node,
+            message:
+              "Import only from baseui's public API. Source code may change in breaking ways without notice.",
```

Comment:
Where is public api defined? May want to refer to one level deep or similar

Reply:
I rewrote this one a few times trying to find the right concise wording. How about:

> Please limit imports to "baseui" or one level deep modules such as "baseui/button". Importing more than one level deep is dangerous as these files are not considered part of our public API.

Reply:
Yeah I think it's better to be more verbose